src/components/breakpointHandler.ts

```diff
@@ -0,0 +1,667 @@
+        });
+        this.setBreakpoints = [];
+    }
+
+    private handleBreakpointAdded(bp: vscode.SourceBreakpoint) {
+        let location = this.pyCppTranslation(
+            bp.location.uri,
+            bp.location.range.start.line
+        );
+
+        if (!location) {
+            console.log("no location found");
+            return;
+        }
+
+        let newBp = this.createBreakpoint(
+            location.line,
+            location.cachePath,
+            location.functionName,
+            location.target
+        );
+
+        const funcName = location.functionName;
+        let alreadyExists = this.setBreakpoints.find((bp) => {
+            return bp.identifier === this.bpIdentifier(newBp, funcName);
+        });
+        if (!alreadyExists) {
+            this.setBreakpoints.push({
+                bp: newBp,
+                identifier: this.bpIdentifier(newBp, funcName)
+            });
+            vscode.debug.addBreakpoints([newBp]);
+        }
+
+    }
+
+    public pyCppTranslation(uri: vscode.Uri, line: number) {
+        // Translate a Python Location to a C++ Location
+        let path = normalizePath(uri.fsPath);
+        if (!path) {
+            return; // Illegal path 
+        }
+
+        // If the path hasn't been added yet, then we'll handle the BP
+        // later on when the path gets added after compilation of the
+        // DaCe program 
+        // If the path has been added, then we'll receive an array of
+        // registered DaCe function names from that path
+        let functions = this.files[path];
+        if (!functions) {
+            return;
+        }
+
+        for (let currentFunc of functions) {
+            let cachePath = currentFunc.cache;
+
+            // Get the corresponding Node, if the line isn't in the map
+            // then we expect it's not part of a DaCe program,
+            // hence we do nothing and return 
+            let nodes = this.getNode(
+                line + 1,
+                cachePath + "/map/map_py.json"
+            );
+            if (!nodes || (nodes && nodes.length == 0))
+                continue;
+
+            let range;
+            for (const node of nodes) {
+                range = getCppRange(
+                    node,
+                    cachePath + "/map/map_cpp.json"
+                )
+                if (range)
+                    break;
+            }
+
+            if (!range || !range.from)
+                continue;
+
+            return {
+                line: range.from,
+                cachePath: cachePath,
+                functionName: currentFunc.name,
+                target: currentFunc.target_name
+            }
+        }
+
+        return undefined;
+    }
+
+    public handleNodeAdded(node: Node, sdfgName: string) {
+        // Creating a Breakpoint for the BP panel
+        /* let uri = vscode.Uri.parse(
+            'C:/Users/Benjamin/Documents/Code/Bachelor/dace/.dacecache/myprogram/program.sdfg'
```

Comment:
Careful, I would remove this..! 😉

Reply:
Ups, this should not be here, removed it